lib/web_ui/dev/test_runner.dart

```diff
@@ -234,13 +249,17 @@ class TestCommand extends Command<bool> {
       ...<String>['-r', 'compact'],
       '--concurrency=$concurrency',
       if (isDebug) '--pause-after-load',
-      '--platform=chrome',
+      '--platform=$browser',
       '--precompiled=${environment.webUiRootDir.path}/build',
       '--',
       ...testFiles.map((f) => f.relativeToWebUi).toList(),
     ];
-    hack.registerPlatformPlugin(<Runtime>[Runtime.chrome], () {
+
+    hack.registerPlatformPlugin(
+        <Runtime>[supportedBrowsers.supportedBrowsersToRuntimeMap[browser]],
+        () {
```

Comment:
This is the place where we select a browser based on the `Runtime`. It feels strange that we do this again inside `BrowserPlatform`. Would it be better to select a `BrowserManager` or some such here once and remove all of the switching code inside the `BrowserPlatform`?

Reply:
There is only one switching code, which is screenshots. But I agree. I will do it in the next PR. (Seems like another 200 lines let me not add more to this one)